### PR TITLE
Improve main.rs

### PR DESCRIPTION
**Pull request by EnokiUN**
This pr includes improving how main.rs handles env::var errors and also changed the require_authentication macro to a function for get_current_user_id

### CODE REVIEW

File: src/info_endpoints.rs [hunk 1]
```diff
@@ -1,19 +1,41 @@
 use crate::board;
-use crate::require_authentication;
 use crate::AppState;
-use actix_web::{get, web, HttpRequest, HttpResponse, Responder};
+use actix_web::web::Data;
+use actix_web::{get, HttpRequest, HttpResponse, Responder, Result};
 use serde_json::json;
 
 #[get("/board")]
-async fn get_board(state: web::Data<AppState>) -> impl Responder {
```
Comment (TAG-Epic) [inline src/info_endpoints.rs]: Please revert

File: src/main.rs [hunk 2]
```diff
@@ -6,14 +6,16 @@ mod account_endpoints;
 mod board;
 mod info_endpoints;
 mod user;
-//#[macro_use] mod macros;
 
 #[tokio::main]
 async fn main() -> Result<(), std::io::Error> {
-    let redis_client = redis::Client::open(env::var("REDIS_URI").unwrap()).unwrap();
+    let redis_client = redis::Client::open(
+        env::var("REDIS_URL").unwrap_or_else(|_| "redis://localhost:6379/1".to_string()),
```
Comment (TAG-Epic) [inline src/main.rs]: No need for /1

File: src/main.rs [hunk 3]
```diff
@@ -6,14 +6,16 @@ mod account_endpoints;
...
+    )
+    .expect("Could not find \"REDIS_URL\" environment variable");
```
Comment (TAG-Epic) [inline src/main.rs]: Isnt this already handled?

**Review by TAG-Epic — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 3 above)
Comment (EnokiUN) [reply src/main.rs]: yes, but I changed it to an `expect` to provide more context

**Pull request merged**